src/components/CampaignInitialisation/CampaignInitialisationScreen.tsx

```diff
@@ -0,0 +1,162 @@
+import { size } from "../../common/styles";
+import { NavigationProps } from "../../types";
+import { UpdateByRestartingContent } from "./UpdateByRestartingContent";
+import { UpdateFromAppStoreContent } from "./UpdateFromAppStoreContent";
+import { LoadingView } from "../Loading";
+import { useUpdateCampaignConfig } from "../../hooks/useUpdateCampaignConfig/useUpdateCampaignConfig";
+import { useAuthenticationContext } from "../../context/auth";
+import { useCheckVersion } from "../../hooks/useCheckVersion/useCheckVersion";
+import { useCheckUpdates } from "../../hooks/useCheckUpdates";
+
+const styles = StyleSheet.create({
+  wrapper: {
+    ...StyleSheet.absoluteFillObject,
+    justifyContent: "center",
+    alignItems: "center",
+    paddingHorizontal: size(5)
+  },
+  content: {
+    width: 512,
+    maxWidth: "100%",
+    marginTop: -size(4)
+  }
+});
+
+const RETRY_UPDATE_TIMES = 3;
+
+export const CampaignInitialisationScreen: FunctionComponent<NavigationProps> = ({
+  navigation
+}) => {
+  useEffect(() => {
+    Sentry.addBreadcrumb({
+      category: "navigation",
+      message: "CampaignInitialisationScreen"
+    });
+  }, []);
+
+  const { token, endpoint } = useAuthenticationContext();
+  const { fetchingState, updateCampaignConfig } = useUpdateCampaignConfig(
+    token,
+    endpoint
+  );
+  const checkVersion = useCheckVersion();
+  const checkUpdates = useCheckUpdates();
+
+  useEffect(() => {
+    updateCampaignConfig();
+    // updating the campaign config should only happen once when this screen is loaded
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const continueToNormalFlow = useCallback(() => {
+    const flowType = navigation.getParam("flowType", "DEFAULT");
+    switch (flowType) {
+      case "DEFAULT":
+        navigation.navigate("CollectCustomerDetailsScreen");
+        break;
+      case "MERCHANT":
+        navigation.navigate("MerchantPayoutScreen");
+        break;
+    }
+  }, [navigation]);
+
+  const [outdatedType, setOutdatedType] = useState<"BINARY" | "BUILD">();
+
+  const setState = useState()[1];
+  /**
+   * Tries up to RETRY_UPDATE_TIMES times to get the latest build version.
+   * If unable to retrieve it, don't allow the user to continue since the user
+   * has already failed the compatibility check.
+   */
+  const getNewBuildIfAny = useCallback(async (): Promise<void> => {
+    let lastUpdateResult;
+    for (let i = 0; i < RETRY_UPDATE_TIMES; i++) {
+      Sentry.addBreadcrumb({
+        category: "checkUpdates",
+        message: `attempt ${i + 1}`
+      });
+      const updateResult = await checkUpdates(true);
+      if (updateResult === "UPDATE_READY") {
+        setOutdatedType("BUILD");
+        return;
+      } else {
+        lastUpdateResult = updateResult;
+        continue;
+      }
+    }
+
+    // In the case that there's NO_UPDATE, then we probably set the
+    // minimum build version too early, before the app updates have
+    // been released.
```

Comment:
Hmm a bit confused about what this means. Could you explain when this might happen from a user's perspective? When we say `NO_UPDATE` do we also include `TOO_RECENT_CHECK` since I see that that can also be returned from `useCheckUpdates`?

Reply:
This will happen in the case when the user has spotty internet and/or offline, so the latest build couldn't be downloaded after the 3 tries.

`useCheckUpdates` won't return `TOO_RECENT_CHECK` in this case because we pass in the `forceCheck` boolean to `checkUpdates`, so that will check for updates regardless of when the last check was.

Hmmm but this raises the issue of what happens if the fetching of the campaign config didn't succeed in the first place. Should we just show the Error boundary? Should we also try a few times? Should there be some exponential backoff for the retries? 🤔 

What do you think?

Reply:
Ohh I understand now. When we are in this part of the code, the premise is that the build is outdated. And yet we see `NO_UPDATE` so that's wrong. Makes sense! 

As for showing error boundary -- if the user's internet is not good in the first place, we are throwing an error from `useCheckVersion` right? Not very familiar with this part of the code but would that already trigger the error boundary?

Anyways I think it might be a good idea to show error boundary if the issue is internet connection, cuz they can't use our app without having internet connection anyway 😆

Reply:
Yea I overlooked the error handling before `useCheckVersion`, so when `updateCampaignConfig`, I didn't handle the error.

Good point about the usage of the rest of the app lol, will handle it by showing the error boundary then!